### PR TITLE
Bug fix. To allow state transition to last state.

### PR DESCRIPTION
**Pull request by rolandtritsch**
Right now you cannot transition to last state.

### CODE REVIEW

File: lib/ecto/fsm.ex [hunk 1]
```diff
@@ -163,7 +163,7 @@ defmodule Ecto.FSM do
       next_states = @to || unquote(next_states)
 
       @fsm Map.put(@fsm, {state_name, trans}, {__MODULE__, next_states})
-      @__states_names__ [state_name | @__states_names__] |> Enum.uniq()
+      @__states_names__ [state_name | @__states_names__] ++ next_states |> Enum.uniq() |> Enum.sort()
```
Comment (rolandtritsch) [inline lib/ecto/fsm.ex]: Not sure, if this is the most elegant way of doing this, but ... the next_states need be taken into consideration.

File: lib/ecto/fsm.ex [hunk 2]
```diff
@@ -163,7 +163,7 @@ defmodule Ecto.FSM do
       next_states = @to || unquote(next_states)
 
       @fsm Map.put(@fsm, {state_name, trans}, {__MODULE__, next_states})
-      @__states_names__ [state_name | @__states_names__] |> Enum.uniq()
+      @__states_names__ [state_name | @__states_names__] ++ next_states |> Enum.uniq()
```
Comment (rolandtritsch) [inline lib/ecto/fsm.ex:166]: Adding next_states (to make sure we catch/get the "last" state).

File: test/ecto/fsm_machine_test.exs [hunk 3]
```diff
@@ -34,7 +34,10 @@ defmodule Ecto.FSM.MachineTest do
     end
 
     test ".find_handler/1", %{locker: s} do
-      assert match?(Locker, Machine.find_handler({s, :one}))
+      # State unlocked is only state with no match all
+      s = %{s | status: :unlocked}
+
+      assert match?(Locker, Machine.find_handler({s, :lock}))
```
Comment (rolandtritsch) [inline test/ecto/fsm_machine_test.exs:40]: Unrelated. Quick fix to make test pass (again).

File: test/ecto/fsm_test.exs [hunk 4]
```diff
@@ -71,4 +71,32 @@ defmodule Ecto.FSM.NotationTest do
       assert match?(%{:goto_s1 => ^mod}, mod.event_bypasses())
     end
   end
+
+
+  describe ".states_names/0" do
```
Comment (rolandtritsch) [inline test/ecto/fsm_test.exs:76]: Test to test state_names/0.

File: test/ecto/fsm_test.exs [hunk 5]
```diff
@@ -71,4 +71,32 @@ defmodule Ecto.FSM.NotationTest do
...
+          transition s1({:goto_s2, nil}, s) do
+            {:next_state, :s2, s}
+          end
+
+          transition s2({:goto_s3, nil}, s) do
+            {:next_state, :s3, s}
+          end
+        end
+      
+      assert match?([:s1, :s2, :s3], mod.states_names() |> Enum.sort())
```
Comment (rolandtritsch) [inline test/ecto/fsm_test.exs:99]: Before the fix this only returned [:s1, :s2].

**Comment by rolandtritsch** [comment]
Hi @jeanparpaillon. What prevents us from merging this?

File: test/ecto/fsm_test.exs (hunk 5 above)
Comment (jeanparpaillon) [reply test/ecto/fsm_test.exs:99]: ok, got it, thanks

**Comment by jeanparpaillon** [comment]
Hi, sorry for the delay, just merged.

**Pull request merged**